eZ/Publish/Core/FieldType/Keyword/KeywordStorage.php

```diff
@@ -26,7 +26,7 @@ class KeywordStorage extends GatewayBasedStorage
      */
     public function storeFieldData(VersionInfo $versionInfo, Field $field, array $context)
     {
-        if (empty($field->value->externalData)) {
+        if (empty($field->value->externalData) && !is_array($field->value->externalData)) {
```

Comment:
looks like the wrong place to fix it, if value is wrong it should be caught higher up.